src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/OleVariant.cs

```diff
@@ -0,0 +1,560 @@
+                if (versionedStream != null && versionedStream->_stream != null)
+                {
+                    Marshal.Release(versionedStream->_stream);
+                }
+                Marshal.FreeCoTaskMem((nint)versionedStream);
+            }
+            else if (VarType == VT_CF)
+            {
+                ClipboardData* clipboardData = _typeUnion._unionTypes.clipboardData;
+                if (clipboardData != null)
+                {
+                    Marshal.FreeCoTaskMem(clipboardData->_data);
+                    Marshal.FreeCoTaskMem((nint)clipboardData);
+                }
+            }
+            else if (VarType.HasFlag(VarEnum.VT_VECTOR))
+            {
+                switch (VarType & ~VarEnum.VT_VECTOR)
+                {
+                    case VarEnum.VT_BSTR:
+                        foreach (var str in GetRawDataRef<Vector<IntPtr>>().AsSpan())
+                        {
+                            Marshal.FreeBSTR(str);
+                        }
+                        break;
+                    case VarEnum.VT_LPSTR:
+                    case VarEnum.VT_LPWSTR:
+                    foreach (var str in GetRawDataRef<Vector<IntPtr>>().AsSpan())
+                        {
+                            Marshal.FreeCoTaskMem(str);
+                        }
+                        break;
+                    case VarEnum.VT_CF:
+                        foreach (var cf in GetRawDataRef<Vector<ClipboardData>>().AsSpan())
+                        {
+                            Marshal.FreeCoTaskMem(cf._data);
+                        }
+                        break;
+                    case VarEnum.VT_VARIANT:
+                        foreach (var variant in GetRawDataRef<Vector<OleVariant>>().AsSpan())
+                        {
+                            variant.Dispose();
+                        }
+                        break;
+                    default:
+                        break;
+                }
+                Marshal.CoTaskMemFree((nint)GetRawDataRef<Vector<byte>>()._data);
+            }
+
+            // Clear out this OleVariant instance.
+            this = default;
+#endif
+        }
+
+#pragma warning disable CS0618 // We support the obsolete CurrencyWrapper type
+        /// <summary>
+        /// Create an <see cref="OleVariant"/> instance from the specified value.
+        /// </summary>
+        /// <typeparam name="T">The type of the specified value.</typeparam>
+        /// <param name="value">The value to wrap in an <see cref="OleVariant"/>.</param>
+        /// <returns>An <see cref="OleVariant"/> that contains the provided value.</returns>
+        /// <exception cref="ArgumentException">When <typeparamref name="T"/> does not directly correspond to a <see cref="VarEnum"/> variant type.</exception>
+        public static OleVariant Create<T>([DisallowNull] T value)
+        {
+            Unsafe.SkipInit(out OleVariant variant);
```

Comment:
Is there a reason we want to declare this this way? It isn't obvious and is borderline for something we should be doing.

Reply:
I did it this way because I felt there's no reason for us to zero-initialize the whole variant type if we're going to set the specific fields that we want to set anyway.

The C# style rules/Roslyn rules we have in the repo suggest that we use the "declare a variable in the out parameter location" pattern if that's the first usage. I can change it if you prefer, but I personally much prefer this declaration pattern in general. I can understand it being confusing in this case though.

Looking through the repo, I see usages of this same pattern in Decimal.DecCalc.cs and throughout much of the Numerics/Intrinsics space. I don't see any usages of a different pattern when in this scenario (where the variable is being declared and not passed in as a byref parameter).

Reply:
Okay.